xchange-stream-binance/src/main/java/info/bitrich/xchangestream/binance/futures/BinanceFuturesStreamingTradeService.java

```diff
@@ -0,0 +1,97 @@
+package info.bitrich.xchangestream.binance.futures;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import info.bitrich.xchangestream.binance.BinanceStreamingTradeService;
+import info.bitrich.xchangestream.binance.BinanceUserDataStreamingService;
+import info.bitrich.xchangestream.binance.dto.BaseBinanceWebSocketTransaction;
+import info.bitrich.xchangestream.binance.dto.ExecutionReportBinanceUserTransaction;
+import info.bitrich.xchangestream.binance.futures.dto.OrderTradeUpdateBinanceUserTransaction;
+import info.bitrich.xchangestream.core.StreamingTradeService;
+import info.bitrich.xchangestream.service.netty.StreamingObjectMapperHelper;
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.processors.FlowableProcessor;
+import io.reactivex.rxjava3.processors.PublishProcessor;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.trade.UserTrade;
+import org.knowm.xchange.exceptions.ExchangeException;
+import org.knowm.xchange.exceptions.ExchangeSecurityException;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+
+public class BinanceFuturesStreamingTradeService implements StreamingTradeService {
```

Comment:
The only difference between BinanceStreamingTradeService and BinanceFuturesStreamingTradeService is the use of OrderTradeUpdateBinanceUserTransaction instead of ExecutionReportBinanceUserTransaction.

Can't we define a generic class BaseBinanceStreamingTradeService\<T>, and define:

class BinanceStreamingTradeService extends  BaseBinanceStreamingTradeService\<ExecutionReportBinanceUserTransaction> {
}

class BinanceFuturesStreamingTradeService extends  BaseBinanceStreamingTradeService\<OrderTradeUpdateBinanceUserTransaction > {
}

This way we don't duplicate code. Single generic base class. Easier to maintain.